Sources_v3/StreamChat/Workers/ChatClientUpdater.swift

```diff
@@ -0,0 +1,162 @@
+                        self.client.connectionStatus = .disconnected(error: nil)
+                    }
+
+                    // Establish web-socket connection automatically if it is required
+                    // by the config. If `shouldConnectAutomatically` is set to `false` it means the
+                    // end-user wants to manually control web-socket connection
+                    // by calling `connect/disconnect` when it makes sense.
+                    if self.client.config.shouldConnectAutomatically {
+                        self.connect(completion: completion)
+                    } else {
+                        completion?(nil)
+                    }
+                } catch {
+                    completion?(error)
+                }
+            case let .failure(error):
+                completion?(error)
+            }
+        }
+    }
+
+    /// Connects the chat client the controller represents to the chat servers.
+    ///
+    /// When the connection is established, `ChatClient` starts receiving chat updates, and `currentUser` variable is available.
+    ///
+    /// - Parameter completion: Called when the connection is established. If the connection fails, the completion is
+    /// called with an error.
+    ///
+    func connect(completion: ((Error?) -> Void)? = nil) {
+        // Connecting is not possible in connectionless mode (duh)
+        guard client.config.isClientInActiveMode else {
+            completion?(ClientError.ClientIsNotInActiveMode())
+            return
+        }
+
+        guard client.connectionId == nil else {
+            log.warning("The client is already connected. Skipping the `connect` call.")
+            completion?(nil)
+            return
+        }
+
+        // Set up a waiter for the new connection id to know when the connection process is finished
+        client.provideConnectionId { [weak client] connectionId in
+            if connectionId != nil {
+                completion?(nil)
+            } else {
+                // Try to get a concrete error
+                if case let .disconnected(error) = client?.webSocketClient?.connectionState {
+                    completion?(ClientError.ConnectionNotSuccessfull(with: error))
+                } else {
+                    completion?(ClientError.ConnectionNotSuccessfull())
+                }
+            }
+        }
+
+        client.webSocketClient?.connect()
+    }
+
+    /// Disconnects the chat client the controller represents from the chat servers. No further updates from the servers
+    /// are received.
+    func disconnect() {
+        // Disconnecting is not possible in connectionless mode (duh)
+        guard client.config.isClientInActiveMode else {
+            log.error(ClientError.ClientIsNotInActiveMode().localizedDescription)
+            return
+        }
+
+        guard client.connectionId != nil else {
+            log.warning("The client is already disconnected. Skipping the `connect` call.")
```

Comment:
`DISconnect call`

Reply:
fixed in 6309a297de5b5882f7a60485741ce64093b57b6a